### PR TITLE
Added GetSupportedOptimizationTypes API and changed the optimizationtype to an enum

### PR DESCRIPTION
**Pull request by prakharsharma10**
In this PR, we have updated the swagger for the API version 2016-10-02. 

The changes include the following:
1. Added GetSupportedOptimizationTypes API
2. Changed the optimization type from a string to an enum

For these changes in the swagger, we do not plan to generate the SDK and write test cases as we'll soon generate the SDK using the 2017-04-02 version of swagger.

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: arm-cdn/2016-10-02/swagger/cdn.json [hunk 1]
```diff
@@ -343,6 +343,47 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cdn/profiles/{profileName}/getSupportedOptimizationTypes": {
```
Comment (fearthecowboy) [inline arm-cdn/2016-10-02/swagger/cdn.json]: Does this return a specific subset of the set in  the `enum` in  `OptimizationType`  ? (ie, it's not just the whole list every time, right?)

File: arm-cdn/2016-10-02/swagger/cdn.json [hunk 2]
```diff
@@ -1861,7 +1915,7 @@
         },
         "optimizationType": {
           "description": "Customer can specify what scenario they want this CDN endpoint to optimize, e.g. Download, Media services. With this information we can apply scenario driven optimization.",
-          "type": "string"
+          "$ref": "#/definitions/OptimizationType"
```
Comment (fearthecowboy) [inline arm-cdn/2016-10-02/swagger/cdn.json]: I'm going to generate the code in the morning to see what exactly the impact will be on the consumer -- but I'm pretty sure that I don't have a problem with this 😄 

I think this is a nice change.

**Review by fearthecowboy — COMMENTED** [review]
Like I said, I'm going to do a quick regen check first thing in the morning, but I think this is fine

File: arm-cdn/2016-10-02/swagger/cdn.json (hunk 1 above)
Comment (prakharsharma10) [inline arm-cdn/2016-10-02/swagger/cdn.json]: Yes, this returns some subset of the list.

File: arm-cdn/2016-10-02/swagger/cdn.json (hunk 2 above)
Comment (prakharsharma10) [inline arm-cdn/2016-10-02/swagger/cdn.json]: Thanks! 👍

**Comment by fearthecowboy** [comment]
Ach!

@prakharsharma10  -- I just noticed that this change is against the `master` branch.

We recently did a re-org in the repo, and changes need to be made against the `current` branch -- can I get you to re-create the PR based off the `current` branch? 

I have to close this PR and you'll have to recreate a new one (I can finish reviewing it then)

**Comment by prakharsharma10** [comment]
Oh! No prob. I was following the instructions [here](https://github.com/Azure/adx-documentation-pr/blob/master/engineering/autorest-to-powershell.md). You can update there too if you want. :)

I have now created the PR against the current branch [here](https://github.com/Azure/azure-rest-api-specs/pull/1523).

**Comment by fearthecowboy** [comment]
Fixed the instructions!  Thanks for the link :D

**Pull request closed without merging**